pydrive2/test/test_file.py

```diff
@@ -733,24 +747,62 @@ def _parallel_uploader(
         )
         self.assertTrue(len(files) == self.FILE_UPLOAD_COUNT)
 
+        return [fi["id"] for fi in upload_files]
+
+    def _parallel_downloader(
+        self, file_ids, num_of_workers, use_per_thread_http=False
+    ):
+        drive = GoogleDrive(self.ga)
+        thread_pool = ThreadPoolExecutor(max_workers=num_of_workers)
+
+        # Create list of gdrive_files.
+        download_files = []
+        for file_id in file_ids:
+            file1 = drive.CreateFile({"id": file_id})
+            file1["title"] = self.getTempFile()
+            download_files.append(file1)
+
+        # Ensure files don't exist yet.
+        for file_obj in download_files:
+            self.assertTrue(not delete_file(file_obj["title"]))
+
+        # Submit upload jobs to ThreadPoolExecutor.
+        futures = []
+        for file_obj in download_files:
+            download_worker = self.DownloadWorker(
+                file_obj, use_per_thread_http
```

Comment:
it should be working w/o `use_per_thread_http` (and DownloadWorker / UploadWorker hopefully don't need that option anymore)